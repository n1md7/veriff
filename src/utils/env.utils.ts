import dotenv from 'dotenv';
dotenv.config();

export class EnvUtils {
  static getServerPort() {
    return Number(process.env.PORT) || 3000;
  }

  static getVeriffBaseUrl() {
    const url = process.env.VERIFF_API_BASE_URL;
    if (!url) throw new Error('"VERIFF_API_BASE_URL" is missing');

    return url;
  }

  static getVeriffSessionUUID() {
    const uuid = process.env.VERIFF_SESSION_UUID;
    if (!uuid) throw new Error('"VERIFF_SESSION_UUID" is missing');

    return uuid;
  }
}
