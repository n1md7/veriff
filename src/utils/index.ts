export * from './string.utils';
export * from './response.utils';
export * from './env.utils';
