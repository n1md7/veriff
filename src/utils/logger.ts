export class Logger {
  private static message(msg: unknown, ctx: string, type: string) {
    console.log(`${type} - ${process.pid} - [${Date.now()}] [${ctx}]: ${JSON.stringify(msg)}`);
  }

  static log(msg: unknown, ctx: string) {
    Logger.message(msg, ctx, 'LOG');
  }

  static error(msg: unknown, ctx: string) {
    Logger.message(msg, ctx, 'ERROR');
  }

  static warn(msg: unknown, ctx: string) {
    Logger.message(msg, ctx, 'WARN');
  }
}
