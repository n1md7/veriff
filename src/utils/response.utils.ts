export class ResponseUtils {
  static getResponse(code, data) {
    const random = Math.floor(Math.random() * 10) + 1;
    if (random > 8) {
      return [500, 'There is something wrong with the service'];
    }

    return [code, data];
  }
}
