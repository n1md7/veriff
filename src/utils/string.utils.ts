import { EnvUtils } from './env.utils';

export const findMatch = (haystack: string, needle = EnvUtils.getVeriffSessionUUID()) => {
  return haystack.includes(needle);
};
