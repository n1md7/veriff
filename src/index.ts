import express from 'express';

import { xRequestId } from './middlewares/x-request-id.middleware';
import { errorHandler } from './middlewares/error.handler.middleware';
import { session } from './routes/sessions.route';
import { Logger } from './utils/logger';
import { EnvUtils } from './utils';

const app = express();
const port = EnvUtils.getServerPort();

app.use(xRequestId());
app.use('/api/sessions', session);
app.use(errorHandler());

app.listen(port, '0.0.0.0', () => {
  Logger.log(`Service is listening at http://localhost:${port}`, 'Init');
});

export { app };
