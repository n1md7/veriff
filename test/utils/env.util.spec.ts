import { EnvUtils } from '../../src/utils';

describe('EnvUtils', () => {
  const VERIFF_API_BASE_URL = 'VERIFF_API_BASE_URL';
  const VERIFF_SESSION_UUID = 'VERIFF_SESSION_UUID';

  beforeEach(() => {
    process.env.VERIFF_API_BASE_URL = VERIFF_API_BASE_URL;
    process.env.VERIFF_SESSION_UUID = VERIFF_SESSION_UUID;
  });

  describe('getVeriffBaseUrl', () => {
    it('should get value', () => {
      expect(EnvUtils.getVeriffBaseUrl()).toBe(VERIFF_API_BASE_URL);
    });
    it('should throw', () => {
      delete process.env.VERIFF_API_BASE_URL;
      expect(() => EnvUtils.getVeriffBaseUrl()).toThrowErrorMatchingSnapshot();
    });
  });

  describe('getVeriffSessionUUID', () => {
    it('should get value', () => {
      expect(EnvUtils.getVeriffSessionUUID()).toBe(VERIFF_SESSION_UUID);
    });
    it('should throw', () => {
      delete process.env.VERIFF_SESSION_UUID;
      expect(() => EnvUtils.getVeriffSessionUUID()).toThrowErrorMatchingSnapshot();
    });
  });

  describe('getServerPort', () => {
    it('should get value', () => {
      expect(EnvUtils.getServerPort()).toBe(3000);
    });
    it('should not throw', () => {
      delete process.env.PORT;
      expect(EnvUtils.getServerPort()).toBe(3000);
      expect(() => EnvUtils.getServerPort()).not.toThrow();
    });
  });
});
