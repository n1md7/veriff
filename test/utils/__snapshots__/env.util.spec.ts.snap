// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnvUtils getVeriffBaseUrl should throw 1`] = `"\\"VERIFF_API_BASE_URL\\" is missing"`;

exports[`EnvUtils getVeriffSessionUUID should throw 1`] = `"\\"VERIFF_SESSION_UUID\\" is missing"`;
